Performance/QPSBenchmark/Sources/QPSBenchmark/Runtime/Async/AsyncWorkerServiceImpl.swift

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2022, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import GRPC
+import NIOCore
+
+// Implementation of the control service for communication with the driver process.
+class AsyncWorkerServiceImpl: Grpc_Testing_WorkerServiceAsyncProvider {
+  let interceptors: Grpc_Testing_WorkerServiceServerInterceptorFactoryProtocol? = nil
+
+  private let finishedPromise: EventLoopPromise<Void>
+  private let serverPortOverride: Int?
+
+  private var runningServer: AsyncQPSServer?
+  private var runningClient: AsyncQPSClient?
```

Comment:
This is not thread safe unfortunately. In this instance we can just turn the `class` into an `actor`.